src/main/java/ui/MenuControl.java

```diff
@@ -129,12 +131,31 @@ public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number
         cols.getItems().addAll(createRight, createLeft, closePanel);
         return cols;
     }
+    
+    private void onBoardSave() {
+        logger.info("Menu: Boards > Save");
+        
+        List<PanelInfo> panels = getCurrentPanels();
+        
+        if (openBoardName.equals("")) {
+            onBoardSaveAs();
+        }
```

Comment:
Should there be a `return` here? Seems like after we go into `onBoardSaveAs` we'd either have saved the board or not, and we shouldn't have to come back here.